shop/src/main/java/ro/msg/learning/shop/configuration/ClassConfig.java

```diff
@@ -0,0 +1,24 @@
+package ro.msg.learning.shop.configuration;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import ro.msg.learning.shop.service.strategy.LocationStrategy;
+import ro.msg.learning.shop.service.strategy.SingleLocationStrategy;
+
+@Configuration
+public class ClassConfig {
+
+    @Bean
+    public LocationStrategy chooseStrategy(@Value("${strategy}")String s) throws Exception {
+        if(s.equals("SingleLocationStrategy")){
```

Comment:
Try using enums for the static values of your strategies. Basically, we would have an enum containing the two values for the strategies (eg. enum LocationStrategy with values SINGLE_LOCATION_STRATEGY and MOST_ABUNDANT_STRATEGY).